lib/registries/config/environment.js

```diff
@@ -37,7 +37,7 @@ module.exports = function(environment) {
             display: 'OSF Registries',
             https: true,
             name: 'OSF',
-            urlRegex: '^https?://(?:.*staging.*\\.)?osf\\.io.*$',
+            urlRegex: '^https?://(?:.*[staging|test].*\\.)?osf\\.io.*$',
```

Comment:
A couple things here:
- Square brackets define character classes, so `[staging|test]` matches a single character. It's the same as `[aeginst|]`. I think you meant `(staging|test)`
- This regex will match `http://staging-share.osf.io/...`, which is part of the original problem. When multiple identifiers match, it might use any:
https://github.com/CenterForOpenScience/ember-osf-web/blob/9a62dc99f5d9908b011e94fe051c39eff79d2cac/lib/registries/addon/services/share-search.ts#L189-L195

```suggestion
            urlRegex: '^https?://(?:.*(?:staging\\d?|test)\\.)?osf\\.io.*$',
```